ui/plugin-system/src/utils/variables.ts

```diff
@@ -57,8 +66,15 @@ export const parseTemplateVariables = (text: string) => {
   let match;
 
   while ((match = regex.exec(text)) !== null) {
-    if (match && match.length > 1 && match[1]) {
-      matches.add(match[1]);
+    if (match) {
+      if (match[1]) {
+        // \$(\w+)\
+        matches.add(match[1]);
+      }
+      if (match[2]) {
```

Comment:
I'd be in favor of adding an `else` here, to be more explicit about the fact that matching both is not possible